simtools/model/mirrors.py

```diff
@@ -136,23 +131,37 @@ def get_single_mirror_parameters(self, number):
 
         Returns
         -------
-        (pos_x, pos_y, diameter, flen, shape): tuple of float
-            X, Y positions, diameter, focal length and shape.
+        (pos_x, pos_y, mirror_diameter, focal_length, shape_type): tuple of float
+            X, Y positions, mirror_diameter, focal length and shape_type.
         """
 
-        if number > self.number_of_mirrors - 1:
-            self._logger.error("Mirror number is out range")
+        def get_mirror_table_mask():
+            if isinstance(self.mirror_table["mirror_id"][0], np.str_):
+                mask = self.mirror_table["mirror_id"] == "id=" + str(number)
+            if isinstance(self.mirror_table["mirror_id"][0], np.int32):
+                mask = self.mirror_table["mirror_id"] == number
+            return mask
+
+        mask = get_mirror_table_mask()
+        if not np.any(mask):
+            self._logger.error(f"Mirror id{number} not in table, using first mirror instead")
+            mask[0] = True
+
+        try:
+            return (
+                self.mirror_table[mask]["mirror_x"].value[0],
+                self.mirror_table[mask]["mirror_y"].value[0],
+                self.mirror_table[mask]["mirror_diameter"].value[0],
+                self.mirror_table[mask]["focal_length"].value[0],
+                self.mirror_table[mask]["shape_type"].value[0],
+            )
+        except KeyError:
+            self._logger.debug("Mirror list missing required column")
             return None
-        return (
-            self._mirrors["pos_x"][number],
-            self._mirrors["pos_y"][number],
-            self._mirrors["diameter"][number],
-            self._mirrors["flen"][number],
-            self._mirrors["shape"][number],
-        )
 
     def plot_mirror_layout(self):
         """
-        Plot the mirror layout (not implemented yet).
+        Plot the mirror layout.
+
+        TODO
```

Comment:
Notice that this is an old version of the code. Probably you started your branch later and then didn't accept this merge. We removed the TODO and raise the error instead. Please revert this change.